pkg/packageinstall/packageinstall.go

```diff
@@ -237,9 +301,35 @@ func (pi *PackageInstallCR) referencedPkgVersion() (datapkgingv1alpha1.Package,
 		}
 	}
 
-	verConfig := verv1alpha1.VersionSelection{Semver: semverConfig}
+	// we only need to populate the versionInfo we know that the packages have constraints that will require this info.
+	v := semver.Version{}
+	if requiresClusterVersion {
+		vi, err := pi.deployFactory.GetClusterVersion(pi.model.Spec.ServiceAccountName, pi.model.Spec.Cluster, &pi.model.ObjectMeta, pi.log)
+		if err != nil {
+			pi.log.Error(err, "Unable to retrieve cluster kubernetes version")
+			return datapkgingv1alpha1.Package{}, err
+		}
+		v, err = dropVersionPreReleaseAndBuild(vi.GitVersion)
+		if err != nil {
+			pi.log.Error(err, "Unable to parse cluster kubernetes version", "version string", vi.GitVersion)
+			return datapkgingv1alpha1.Package{}, err
+		}
+	}
+	k8sConstraint := func(pkgVer string) bool {
+		pkg := versionToPkg[pkgVer]
+		return pi.clusterVersionConstraintsSatisfied(&pkg, v)
+	}
+	kcConstraint := func(pkgVer string) bool {
+		pkg := versionToPkg[pkgVer]
+		return pi.kcVersionConstraintsSatisfied(&pkg)
+	}
 
-	selectedVersion, err := versions.HighestConstrainedVersion(versionStrs, verConfig)
+	vcc := []versions.ConstraintCallback{
+		{Constraint: k8sConstraint, Name: "kubernetes-version-check"},
```

Comment:
should this constraint be only conditionally added when requiresClusterVersion is true? i dunno the impact of running clusterVersionConstraintsSatisfied with a blank version.

Reply:
I guess the only consequence is that we will always have a constraint showing in the error message for the k8s version check, even if the user never specified that constraint,

Reply:
I added a test to show this Test_Package_ConstraintNotGiven_ErrorDoesNotContainMessage